consensus/ibft/state.go

```diff
@@ -97,12 +98,15 @@ func (c *currentState) setState(s IbftState) {
 }
 
 // NumValid returns the number of required messages
-func (c *currentState) NumValid() int {
-	// According to the IBFT spec, the number of valid messages
-	// needs to be 2F + 1
-	// The 1 missing from this equation is accounted for elsewhere
-	// (the current node tallying the messages will include its own message)
-	return 2 * c.validators.MaxFaultyNodes()
+func NumValid(validators ValidatorSet) int {
```

Comment:
Why did you change from method to function? I think currentState or ValidattorSet should have this

Reply:
The reason why I thought is that you can't change calculations without rewriting the code. You just create new struct and method when you want to change calc if you define it as method

Reply:
Good point 👍 

[Refactored](https://github.com/0xPolygon/polygon-edge/pull/513/commits/44c154be5602fe978d70f9f64ad40a84ccf257d6)